breaker.go

```diff
@@ -372,7 +372,9 @@ func (cb *CircuitBreaker) Counters() Counters {
 
 // Reset resets cb's state with StateClosed.
 func (cb *CircuitBreaker) Reset() {
+	cb.mu.Lock()
 	cb.cnt.reset()
+	cb.mu.Unlock()
 	cb.SetState(StateClosed)
```

Comment:
`cb.SetState` also locks using cb.mu. To protect the whole Reset by single lock, could you use `cb.setState(&stateClosed{})` instead of `cb.SetState` like below?

```go
cb.mu.Lock()
defer cb.mu.Unlock()
cb.cnt.reset()
cb.setState(&stateClosed{})
```

Reply:
Hi @matope, done!